.github/workflows/ci.yml

```diff
@@ -0,0 +1,33 @@
+name: CI
+on:
+  push:
+    branches:
+      - "**"
+
+jobs:
+  lint-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Lint commit message
+        uses: wagoid/commitlint-github-action@v4
+  main:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      # Cache node_modules
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 20
+          cache: "npm"
+      - run: npm ci
+      - uses: nrwl/nx-set-shas@v3
+      # This line is needed for nx affected to work when CI is running on a PR
+      - run: git branch --track dev origin/dev
+
+      - run: npm run lint-projects:affected
```

Comment:
Doesn't bundling all the jobs into one means if one fails, you need to open the job to see which one?

Reply:
Also, while we might want to only run CI on staged files locally and at commit time? Is this even faster, given that all the CI jobs can be parallelised?